TTCN3_Project/CANFilterControl/com/testingtech/playits/canfilter/Car2XControlExternalFunctionProvider.java

```diff
@@ -0,0 +1,97 @@
+package com.testingtech.playits.canfilter;
+
+import java.io.IOException;
+
+import org.etsi.ttcn.tci.BooleanValue;
+import org.etsi.ttcn.tci.CharstringValue;
+import org.etsi.ttcn.tci.IntegerValue;
+
+import com.testingtech.ttcn.annotation.ExternalFunction;
+import com.testingtech.ttcn.tri.AnnotationsExternalFunctionPlugin;
+import com.testingtech.util.UUID;
+
+@ExternalFunction.Definitions(Car2XControlExternalFunctionProvider.class)
+public class Car2XControlExternalFunctionProvider extends
+    AnnotationsExternalFunctionPlugin {
+
+  private static Process process;
+  private static String processId;
+
+  /**
+   * Starts a CAN filter service process with specified host and port
+   * 
+   * @param host
+   *            host name
+   * @param portNumber
+   *            port
+   * @return process id
+   */
+  @ExternalFunction(name = "start_Filter", module = "Car2X_Control")
+  public CharstringValue start_Filter(CharstringValue host,
+      IntegerValue portNumber) {
+    return start_Filter(
```

Comment:
One generally **does not** put a method after the return keyword, since this this makes debugging a lot harder. But you really have to explain, why it is necessary to have two methods with same name but different access modifier. That does not look like the proper way of method overloading.